Products/ZenModel/Device.py

```diff
@@ -400,12 +400,36 @@ def getRRDTemplates(self):
         """
         if not hasattr(self, 'zDeviceTemplates'):
             return ManagedEntity.getRRDTemplates(self)
-        result = []
-        for name in self.zDeviceTemplates:
-            template = self.getRRDTemplateByName(name)
+        templates = []
+        for templateName in self.zDeviceTemplates:
+            if templateName.endswith('-replacement') or \
+                    templateName.endswith('-addition'):
+                continue
+
+            template = self.getRRDTemplateByName(templateName)
+            replacement = self.getRRDTemplateByName(
+                '{}-replacement'.format(templateName))
+
+            if replacement and replacement not in templates:
+                templates.append(replacement)
+                #self.setZenProperty(
+                #    'zDeviceTemplates',
+                #    self.zDeviceTemplates + [replacement.titleOrId()])
```

Comment:
Commented out code should be removed.  It's not doing anything and doesn't help explain (to me) the surrounding code.